container-disc/src/main/java/com/yahoo/container/jdisc/component/Deconstructor.java

```diff
@@ -35,13 +39,29 @@
 
     private static final Logger log = Logger.getLogger(Deconstructor.class.getName());
 
+    private static final Duration SHUTDOWN_DECONSTRUCT_TIMEOUT = Duration.ofMinutes(10);
+
+    public enum Mode {
+        RECONFIG,  // Delay deconstruction to allow old components to finish processing in-flight requests.
+        SHUTDOWN   // The container is shutting down. Start deconstructing immediately, and wait until all components
+                   // are deconstructed, to prevent shutting down while deconstruct is in progress.
+    }
+
+    // TODO: make private again
```

Comment:
Yes, was made visible for testing. All tests can use the synchronous shutdown instead :)